### DISCUSSION TITLE
Weird behavior after googletest ends

### DISCUSSION THREAD

**Discussion by iljamaki** [zeux/pugixml#547] [Q&A] [answered]
I have Windows 10 Pro and Visual Studio 2022. My project uses googletest library for unit tests. I recently started using pugixml (version v1.13) as a header-only utility in the project and when I wrote a unit test for a new feature that uses pugixml I noticed that when the execution flow of the test has already left the unit test function body, some pugixml related log entries still keep appearing in the test log, which takes several minutes (at least when run via Test Explorer and when the test has made pugixml parse HTML input over 13000 lines long) and keeps the test framework busy and it cannot start running the next test. I wonder what is going on, and whether there would be something I could do to get rid of that. At least I hope this is only related to unit tests (maybe pugixml somehow detects that googletest framework is running and thinks it's being tested itself?) because if some sort of pugixml cleanup or whatever it's doing is always going to take this long, then there is no chance I can use it in my project.

Here's an extract of the log entries I mean (private information censored with asterisks):
```
R=3 C=1: Have node Show_Node

Returning doctype node <html> stream

R=4 C=1: Returning starttag node <html> stream

Entering ParseHTML...

R=5 C=1: Returning starttag node <head id="**********"> stream

Enter ParseHead...

R=5 C=24: Returning starttag node <meta charset="utf-8"> stream

R=5 C=48: Returning starttag node <title> stream

R=6 C=9: Returning text TextNode [ ***************** ]19 stream

R=7 C=1: Returning endtag node <title> stream

```

**Comment by zeux** [comment, marked as answer]
I'm fairly certain this has nothing to do with googletest. pugixml is not googletest-aware, doesn't use threads, and doesn't use any odd control flow that could lead to code running after the test has finished.

**Reply by iljamaki** [reply]
Thank you for the reply. It turned out the log entries are not produced by pugixml but another source code library that prepares the HTML data for pugixml to parse. Apologies for suspecting the wrong software.
